From b48a3c6f77108c70fb15670513b086a0cae6d1c4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20Erdmann?= <dywi@mailerd.de>
Date: Mon, 25 Aug 2014 23:34:46 +0200
Subject: [PATCH 6/6] tlp-functions: add $DEFAULT_DISK_IO_SCHEDULER

Make the default io scheduler a constant rather than hardcoding it in
set_disk_io_sched().
---
 tlp-functions | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/tlp-functions b/tlp-functions
index 604955c..604027e 100755
--- a/tlp-functions
+++ b/tlp-functions
@@ -54,6 +54,7 @@ readonly BAYSTATEFILE=$STATEDIR/bay-saved
 
 readonly DISK_NOP_WORDS="_ keep"
 readonly DEFAULT_DISK_DEVICES="sda"
+readonly DEFAULT_DISK_IO_SCHEDULER="cfq"
 
 # ----------------------------------------------------------------------------
 # Functions
@@ -748,8 +749,8 @@ set_disk_io_sched () { # set disk io scheduler
     #
     for dev in $DISK_DEVICES; do
         get_disk_dev $dev
-        # get sched from argv, add cfq as default when list is too short
-        sched=${1:-cfq}
+        # get sched from argv, use default scheduler when list is too short
+        sched=${1:-${DEFAULT_DISK_IO_SCHEDULER}}
         schedctrl="/sys/block/$disk_dev/queue/scheduler"
         log_message="set_disk_io_sched: $disk_dev [$disk_id] $sched"
 
-- 
2.0.4

